src/ObjectUtils/PdbFileLlvm.cpp

```diff
@@ -36,20 +47,72 @@ class SymbolInfoVisitor : public llvm::codeview::SymbolVisitorCallbacks {
     SymbolInfo symbol_info;
     symbol_info.set_name(proc.Name.str());
     symbol_info.set_demangled_name(llvm::demangle(proc.Name.str()));
+
+    // The ProcSym's name does not contain an argument list. However, this information is required
+    // when dealing with overloads and it is available in the type info stream. See:
+    // https://llvm.org/docs/PDB/TpiStream.html
+    llvm::StringRef argument_list = RetrieveArgumentList(proc);
+    if (!argument_list.empty()) {
+      symbol_info.set_demangled_name(
+          absl::StrCat(symbol_info.demangled_name(), argument_list.data()));
+    }
+
     // The address in PDB files is a relative virtual address (RVA), to make the address compatible
     // with how we do the computation, we need to add both the load bias (ImageBase for COFF) and
     // the offset of the executable section.
     symbol_info.set_address(proc.CodeOffset + object_file_info_.load_bias +
                             object_file_info_.executable_segment_offset);
     symbol_info.set_size(proc.CodeSize);
+
     *(module_symbols_->add_symbol_infos()) = std::move(symbol_info);
 
     return llvm::Error::success();
   }
 
  private:
+  [[nodiscard]] llvm::StringRef RetrieveArgumentList(const llvm::codeview::ProcSym& proc) const {
+    llvm::codeview::LazyRandomTypeCollection& type_collection = type_info_stream_->typeCollection();
+
+    // We expect function types being either LF_PROCEDURE or LF_MFUNCTION, which are non-simple
+    // types. However, there are cases where the function type is "<no type>. In those cases, we
```

Comment:
So, "no type" is the same as "isSimple()"?

Reply:
Not exactly. Simple types also contain base types etc. But "no type" should be the only simple type occurring as a FunctionType.

The reason why we check for simple types here, is that the `type_info_stream_->getType(proc.FunctionType)` asserts non-simple types as arguments.

Reply:
Clarified that comment.